// asana/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient.ApiClient;
  }
});
Object.defineProperty(exports, "AttachmentsApi", {
  enumerable: true,
  get: function get() {
    return _AttachmentsApi.AttachmentsApi;
  }
});
Object.defineProperty(exports, "AuditLogAPIApi", {
  enumerable: true,
  get: function get() {
    return _AuditLogAPIApi.AuditLogAPIApi;
  }
});
Object.defineProperty(exports, "BatchAPIApi", {
  enumerable: true,
  get: function get() {
    return _BatchAPIApi.BatchAPIApi;
  }
});
Object.defineProperty(exports, "CustomFieldSettingsApi", {
  enumerable: true,
  get: function get() {
    return _CustomFieldSettingsApi.CustomFieldSettingsApi;
  }
});
Object.defineProperty(exports, "CustomFieldsApi", {
  enumerable: true,
  get: function get() {
    return _CustomFieldsApi.CustomFieldsApi;
  }
});
Object.defineProperty(exports, "EventsApi", {
  enumerable: true,
  get: function get() {
    return _EventsApi.EventsApi;
  }
});
Object.defineProperty(exports, "GoalRelationshipsApi", {
  enumerable: true,
  get: function get() {
    return _GoalRelationshipsApi.GoalRelationshipsApi;
  }
});
Object.defineProperty(exports, "GoalsApi", {
  enumerable: true,
  get: function get() {
    return _GoalsApi.GoalsApi;
  }
});
Object.defineProperty(exports, "JobsApi", {
  enumerable: true,
  get: function get() {
    return _JobsApi.JobsApi;
  }
});
Object.defineProperty(exports, "MembershipsApi", {
  enumerable: true,
  get: function get() {
    return _MembershipsApi.MembershipsApi;
  }
});
Object.defineProperty(exports, "OrganizationExportsApi", {
  enumerable: true,
  get: function get() {
    return _OrganizationExportsApi.OrganizationExportsApi;
  }
});
Object.defineProperty(exports, "PortfolioMembershipsApi", {
  enumerable: true,
  get: function get() {
    return _PortfolioMembershipsApi.PortfolioMembershipsApi;
  }
});
Object.defineProperty(exports, "PortfoliosApi", {
  enumerable: true,
  get: function get() {
    return _PortfoliosApi.PortfoliosApi;
  }
});
Object.defineProperty(exports, "ProjectBriefsApi", {
  enumerable: true,
  get: function get() {
    return _ProjectBriefsApi.ProjectBriefsApi;
  }
});
Object.defineProperty(exports, "ProjectMembershipsApi", {
  enumerable: true,
  get: function get() {
    return _ProjectMembershipsApi.ProjectMembershipsApi;
  }
});
Object.defineProperty(exports, "ProjectStatusesApi", {
  enumerable: true,
  get: function get() {
    return _ProjectStatusesApi.ProjectStatusesApi;
  }
});
Object.defineProperty(exports, "ProjectTemplatesApi", {
  enumerable: true,
  get: function get() {
    return _ProjectTemplatesApi.ProjectTemplatesApi;
  }
});
Object.defineProperty(exports, "ProjectsApi", {
  enumerable: true,
  get: function get() {
    return _ProjectsApi.ProjectsApi;
  }
});
Object.defineProperty(exports, "RulesApi", {
  enumerable: true,
  get: function get() {
    return _RulesApi.RulesApi;
  }
});
Object.defineProperty(exports, "SectionsApi", {
  enumerable: true,
  get: function get() {
    return _SectionsApi.SectionsApi;
  }
});
Object.defineProperty(exports, "StatusUpdatesApi", {
  enumerable: true,
  get: function get() {
    return _StatusUpdatesApi.StatusUpdatesApi;
  }
});
Object.defineProperty(exports, "StoriesApi", {
  enumerable: true,
  get: function get() {
    return _StoriesApi.StoriesApi;
  }
});
Object.defineProperty(exports, "TagsApi", {
  enumerable: true,
  get: function get() {
    return _TagsApi.TagsApi;
  }
});
Object.defineProperty(exports, "TaskTemplatesApi", {
  enumerable: true,
  get: function get() {
    return _TaskTemplatesApi.TaskTemplatesApi;
  }
});
Object.defineProperty(exports, "TasksApi", {
  enumerable: true,
  get: function get() {
    return _TasksApi.TasksApi;
  }
});
Object.defineProperty(exports, "TeamMembershipsApi", {
  enumerable: true,
  get: function get() {
    return _TeamMembershipsApi.TeamMembershipsApi;
  }
});
Object.defineProperty(exports, "TeamsApi", {
  enumerable: true,
  get: function get() {
    return _TeamsApi.TeamsApi;
  }
});
Object.defineProperty(exports, "TimePeriodsApi", {
  enumerable: true,
  get: function get() {
    return _TimePeriodsApi.TimePeriodsApi;
  }
});
Object.defineProperty(exports, "TimeTrackingEntriesApi", {
  enumerable: true,
  get: function get() {
    return _TimeTrackingEntriesApi.TimeTrackingEntriesApi;
  }
});
Object.defineProperty(exports, "TypeaheadApi", {
  enumerable: true,
  get: function get() {
    return _TypeaheadApi.TypeaheadApi;
  }
});
Object.defineProperty(exports, "UserTaskListsApi", {
  enumerable: true,
  get: function get() {
    return _UserTaskListsApi.UserTaskListsApi;
  }
});
Object.defineProperty(exports, "UsersApi", {
  enumerable: true,
  get: function get() {
    return _UsersApi.UsersApi;
  }
});
Object.defineProperty(exports, "WebhooksApi", {
  enumerable: true,
  get: function get() {
    return _WebhooksApi.WebhooksApi;
  }
});
Object.defineProperty(exports, "WorkspaceMembershipsApi", {
  enumerable: true,
  get: function get() {
    return _WorkspaceMembershipsApi.WorkspaceMembershipsApi;
  }
});
Object.defineProperty(exports, "WorkspacesApi", {
  enumerable: true,
  get: function get() {
    return _WorkspacesApi.WorkspacesApi;
  }
});
var _ApiClient = require("./ApiClient");
var _AttachmentsApi = require("./api/AttachmentsApi");
var _AuditLogAPIApi = require("./api/AuditLogAPIApi");
var _BatchAPIApi = require("./api/BatchAPIApi");
var _CustomFieldSettingsApi = require("./api/CustomFieldSettingsApi");
var _CustomFieldsApi = require("./api/CustomFieldsApi");
var _EventsApi = require("./api/EventsApi");
var _GoalRelationshipsApi = require("./api/GoalRelationshipsApi");
var _GoalsApi = require("./api/GoalsApi");
var _JobsApi = require("./api/JobsApi");
var _MembershipsApi = require("./api/MembershipsApi");
var _OrganizationExportsApi = require("./api/OrganizationExportsApi");
var _PortfolioMembershipsApi = require("./api/PortfolioMembershipsApi");
var _PortfoliosApi = require("./api/PortfoliosApi");
var _ProjectBriefsApi = require("./api/ProjectBriefsApi");
var _ProjectMembershipsApi = require("./api/ProjectMembershipsApi");
var _ProjectStatusesApi = require("./api/ProjectStatusesApi");
var _ProjectTemplatesApi = require("./api/ProjectTemplatesApi");
var _ProjectsApi = require("./api/ProjectsApi");
var _RulesApi = require("./api/RulesApi");
var _SectionsApi = require("./api/SectionsApi");
var _StatusUpdatesApi = require("./api/StatusUpdatesApi");
var _StoriesApi = require("./api/StoriesApi");
var _TagsApi = require("./api/TagsApi");
var _TaskTemplatesApi = require("./api/TaskTemplatesApi");
var _TasksApi = require("./api/TasksApi");
var _TeamMembershipsApi = require("./api/TeamMembershipsApi");
var _TeamsApi = require("./api/TeamsApi");
var _TimePeriodsApi = require("./api/TimePeriodsApi");
var _TimeTrackingEntriesApi = require("./api/TimeTrackingEntriesApi");
var _TypeaheadApi = require("./api/TypeaheadApi");
var _UserTaskListsApi = require("./api/UserTaskListsApi");
var _UsersApi = require("./api/UsersApi");
var _WebhooksApi = require("./api/WebhooksApi");
var _WorkspaceMembershipsApi = require("./api/WorkspaceMembershipsApi");
var _WorkspacesApi = require("./api/WorkspacesApi");
